### PR TITLE
Add `rb_undef_method`

### PR DESCRIPTION
**Pull request by wagenet**
Is it a macro? NO

Is there a macro version? NO

Defined in:

* 2.3 https://github.com/ruby/ruby/blob/v2_3_7/include/ruby/ruby.h#L1697
* 2.4 https://github.com/ruby/ruby/blob/v2_4_4/include/ruby/ruby.h#L1717
* 2.5 https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L1719
* 2.6 https://github.com/ruby/ruby/blob/v2_6_0_preview2/include/ruby/ruby.h#L1736

### CODE REVIEW

File: src/ruby.rs [hunk 1]
```diff
@@ -741,4 +742,57 @@ tests! {
...
+        unsafe { rb_undef_method(class, cstr!("to_s")) };
+
+        assert.rb_eq(
+            lazy_eval("::TestUndefClass.new.respond_to?(:to_s)"),
+            unsafe { Qfalse }
+        );
+    }
+
+       #[test]
+    fn test_undef_module_method(assert: &mut Assertions) {
```
Comment (chancancode) [inline src/ruby.rs]: Indentation looks weird around here

Comment (chancancode) [inline src/ruby.rs]: Once we have a way to `include` a module from Rust, I think we can do something similar to the above instead

File: src/ruby.rs [hunk 2]
```diff
@@ -741,4 +742,57 @@ tests! {
...
+                rb_cObject
+            )
+        };
+
+        unsafe { rb_undef_method(class, cstr!("to_s")) };
+
+        assert.rb_eq(
+            lazy_eval("::TestUndefClass.new.respond_to?(:to_s)"),
+            unsafe { Qfalse }
+        );
```
Comment (chancancode) [inline src/ruby.rs]: Should we also check calling the method gives you a NoMethodError? Something like

```
assert.rb_eq(
  lazy_eval(r#"
    exception = begin
      TestUndefClass.new.to_s
    rescue => e
      e
    end

    exception.class
  "),
  rb_eNoMethodError
)
```

There are probably terser ways to write it, but that's the idea

File: src/ruby.rs (hunk 1 above)
Comment (wagenet) [reply src/ruby.rs]: Whoops

**Pull request merged**